### ISSUE TITLE
More PositionPlacement weirdness

### ISSUE THREAD

**Issue by afhp-2020** [fvwmorg/fvwm3#115]
I was exploring and ready to report another issue with window placement when I thought I should make sure I run the latest code before re-testing. *That* report will have to wait (unless related to this one, I guess).

So, pulling today's `master` and running a clean make, I seemingly lost my MainPanel.

Looking into it further, it would display outside of the screen with my regular configuration.

So, initial setup:

Primary is 1200px high, offset downwards from the global display which is 1440px high (the primary and the other screen are aligned at the bottom).

`StartupFunc` launches the panel with `+ I Module FvwmButtons MainPanel -g +0+521`.
Applied styles are 
```
Style *Panel StaysOnTop, !Title, !Handles, !Borders, Sticky
Style *Panel WindowListSkip, CirculateSkip
Style MainPanel StartsOnScreen $[monitor.primary]
```
When Fvwm starts, the panel is nowhere to be seen. After activating the log, `Next (MainPanel) Echo $[w.x] $[w.y]` at the FvwmConsole gives `CMD_Echo: -123222132 2`.
Another run gives me `CMD_Echo: 158454410 523`, so it would seem the coordinates are random.

(Note: running `Restart` at the FvwmConsole between each change).

Changing the `StartFunc` call with `-g +0+0` gives the same result with `CMD_Echo: -1196497509 2`.

Changing the `StartFunc` call and removing the geometry specification entirely puts the panel at absolute screen `0, 0`, making it partially visible (asymetrical displays).
Modifying the style to `Style MainPanel StartsOnScreen $[monitor.primary], !WindowListSkip, !CirculateSkip, PositionPlacement 0 0` gives logically the same result: panel placed at `0,0`.
Modifying the style again with `PositionPlacement 0 521p` has no effect, the panel still comes up at `0, 0`.

Combining explicit geometry and PositionPlacement start giving positive results: with the original `-g +0+521` and `PositionPlacement 0 0`, the panel appears at `0 761`, which is offset from the desired vertical middle of the primary by its offset relative to the global screen (240px).
With `StartFunc` launch specifying `-g +0+281` and `PositionPlacement 0 0` the panel is correctly placed in the vertical middle of the primary, which was the original position with `-g +0+521` and no `PositionPlacement`.

So, to summarize:
* explicit geometry is ignored
* explicit coordinates in `PositionPlacement` are ignoed
* explicit geometry combined with `PositionPlacement 0 0` give the proper position.

My initial issue was trying to use `PositionPlacement $[wa.x] $[monitor.$[monitor.primary].y` which would also put the window to some random distant location, it's very likely to be related.

**Comment by ThomasAdam** [comment]
Hi @afhp-2020,

I presume `MainPanel` is an `FvwmButtons` instance?  Can you bundle your entire fvwm config with this ticket, so I can take a look?

Thanks,
Thomas

**Comment by afhp-2020** [comment]
Hello Thomas,

You presume right, `MainPanel` is indeed an FvwmButtons instance (with subpanels, hence the name). It is in part built dynamically but I don't believe that's relevant.

A version of my config is already visible here:
https://github.com/fvwmorg/fvwm3/issues/84#issuecomment-633010079
The only relevant changes are those described above in this issue (the config in #84 is the starting point above).

Regards,
AP

**Comment by ThomasAdam** [comment]
Hi @afhp-2020,

I'm presuming this will have been broken for you in `fvwm2` since you'll need:

`Style MainPanel !USeUSPosition`

... to allow the likes of `PositionPlacement` to work correctly.

That said, I believe there to be a bug with `StartsOnScreen` handling, which I've pushed to `ta/gh-115` -- please try that with the aforementioned style tweak to your config, and I'd be interested to know how you get on.

**Comment by afhp-2020** [comment]
Hi, 

Just ran a quick test with gh-115 (not much time this evening). With that branch, the panel does not open outside the view.

I'm not very clear on what `UseUSPosition` (or its negation) does, but here are the results.

With `Style MainPanel !UseUSPosition`:

* `StartFunc`: no geometry specified
MainPanel Style: no `PositionPlacement` specified
→ panel opens at 0,0 relative to ewmh work area
* `StartFunc`: no geometry specified
MainPanel Style: `PositionPlacement 0p 521p`
→ panel opens at 0,521 within ewmh work area (x offset 48px from absolute)
* `StartFunc`: launch with `-g +0+281`
Style: no `PositionPlacement`
→ panel opens at 72,240 absolute (x offset 24px within work area)

Without `!UseUSPosition` in MainPanel style:
* `StartFunc`: no geometry
Style: no `PositionPlacement`
→ panel opens at 0,0 absolute (partially outside of primary, that's normal). Ewmh struts are applied after the panel is placed.
* `StartFunc`: no geometry
Style: `PositionPlacement 0p 521p`
→ panel opens at 0,0 absolute
* `StartFunc`: launch with `-g +0+521`
Style: no `PositionPlacement`
→ panel opens at the expected position (0,521 absolute, struts applied after panel)
* `StartFunc`: launch with `-g +0+521`
Style: `PositionPlacement 0p 521p`
→ panel opens at 0,521 within work area (correct y=521px, x offset from absolute by struts).

Hope this makes sense.

**Comment by ThomasAdam** [comment]
Hi @afhp-2020,

Yes, it all makes sense. You'll find the same behaviour with `fvwm2` as well.

So this seems fixed to me.

Kindly,
Thomas

**Comment by afhp-2020** [comment]
Agreed.

I'll have to re-read and digest all the placement options at some point.

Regards, 
AP
